test/controllers/incomeSources/cease/IncomeSourceCeasedBackErrorControllerSpec.scala

```diff
@@ -0,0 +1,89 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.incomeSources.cease
+
+import config.featureswitch.IncomeSources
+import config.{AgentItvcErrorHandler, ItvcErrorHandler}
+import controllers.predicates.{NavBarPredicate, SessionTimeoutPredicate}
+import enums.IncomeSourceJourney.{ForeignProperty, IncomeSourceType, SelfEmployment, UkProperty}
+import mocks.controllers.predicates.{MockAuthenticationPredicate, MockIncomeSourceDetailsPredicate}
+import mocks.services.MockSessionService
+import org.scalatest.Assertion
+import play.api.mvc.MessagesControllerComponents
+import play.api.test.Helpers.{defaultAwaitTimeout, status}
+import testUtils.TestSupport
+import views.html.incomeSources.cease.IncomeSourceCeasedBackError
+
+class IncomeSourceCeasedBackErrorControllerSpec extends TestSupport with MockAuthenticationPredicate with MockIncomeSourceDetailsPredicate with MockSessionService {
+
+  object TestIncomeSourceCeasedBackErrorController extends IncomeSourceCeasedBackErrorController(
+    app.injector.instanceOf[SessionTimeoutPredicate],
+    MockAuthenticationPredicate,
+    mockAuthService,
+    MockIncomeSourceDetailsPredicate,
+    mockIncomeSourceDetailsService,
+    app.injector.instanceOf[NavBarPredicate],
+    sessionService = mockSessionService,
+    app.injector.instanceOf[IncomeSourceCeasedBackError]
+  )(appConfig,
+    mcc = app.injector.instanceOf[MessagesControllerComponents],
+    ec,
+    app.injector.instanceOf[ItvcErrorHandler],
+    app.injector.instanceOf[AgentItvcErrorHandler]) {
+
+    def setupOKTest(isAgent: Boolean, incomeSourceType: IncomeSourceType): Assertion = {
+      disableAllSwitches()
+      enable(IncomeSources)
+      setupMockAuthorisationSuccess(isAgent)
+      mockUKPropertyIncomeSourceWithLatency2024()
+
+      val result = (isAgent, incomeSourceType) match {
+        case (true, SelfEmployment) => TestIncomeSourceCeasedBackErrorController.showAgent(incomeSourceType)(fakeRequestConfirmedClient())
+        case (false, SelfEmployment) => TestIncomeSourceCeasedBackErrorController.show(incomeSourceType)(fakeRequestWithActiveSession)
+        case (true, UkProperty) => TestIncomeSourceCeasedBackErrorController.showAgent(incomeSourceType)(fakeRequestConfirmedClient())
+        case (false, UkProperty) => TestIncomeSourceCeasedBackErrorController.show(incomeSourceType)(fakeRequestWithActiveSession)
+        case (true, ForeignProperty) => TestIncomeSourceCeasedBackErrorController.showAgent(incomeSourceType)(fakeRequestConfirmedClient())
+        case (false, ForeignProperty) => TestIncomeSourceCeasedBackErrorController.show(incomeSourceType)(fakeRequestWithActiveSession)
+      }
```

Comment:
Should fix:
Here you could simplify to an if/else isAgent statement, passing in the incomeSource type as the 2nd parameter